dask/dataframe/tests/test_arithmetics_reduction.py

```diff
@@ -869,7 +869,10 @@ def test_reductions_out(frame, axis, out, redfunc):
     if out is not None:
         dsk_out = dd.from_pandas(out, 3)
 
-    np_redfunc = getattr(np, redfunc)
+    # `product` is deprecated as of NumPy 1.25.0, and will be removed in NumPy 2.0.
+    # Please use `prod` instead
+    np_functions = {"product": "prod"}
```

Comment:
This makes it so that we're not actually testing `product` anymore. Instead, how about adding a `_numpy_125` variable similiar to this one

https://github.com/dask/dask/blob/0ea6c0826e8e0ac6c506a06c58b7e047407002e6/dask/array/numpy_compat.py#L11

and using `with pytest.warns(...):` accordingly?

Reply:
We're not calling "np.product", but we're still calling the Dask version.

Reply:
Hrm, it looks like we're calling `np.product` with a Dask DataFrame as input [here](https://github.com/dask/dask/pull/10040/files/e92ee0eb7b50e54da500089b3135ba5b131dc7a8#diff-af8db266da618392c022c8834d360ce244a25316acb7718747822b7ea31aa5e9R885) -- though I could be missing something

Reply:
Right, maybe I don't completely understand the point of line 885. We want to make sure that `numpy` reductions work on Dask dataframes? But then what's the point of testing a deprecated one? We can suppress a warning for now, and then things will break with `numpy` 2.0.

Reply:
Also yes, we were calling `np.product` there, but now we're calling `np.prod` instead, this is what my change does.

Reply:
> We want to make sure that numpy reductions work on Dask dataframes?

Yeah, that's correct

> But then what's the point of testing a deprecated one?

It'd deprecated, but not removed yet. So it'd be good to continue to test that it works as expected. 

> things will break with numpy 2.0

When that happens, our `upstream` build will start to fail and we'll add the appropriate logic to stop testing `np.product` (since it won't exist anymore)

Reply:
Ok, that makes sense.